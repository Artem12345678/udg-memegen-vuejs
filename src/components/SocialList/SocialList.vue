<template>
  <ul class="social">
    <li v-for="(item, index) in links" :key="index" class="social__item">
      <SocialListItem :url="item.url" :icon="item.icon"/>
    </li>
  </ul>
</template>

<script>
import SocialListItem from '@/components/SocialListItem/SocialListItem.vue';

export default {
  name: 'SocialList',
  components: {
    SocialListItem,
  },
  data() {
    return {
      links: [
        {
          url: 'https://www.github.com/Artem12345678',
          icon: 'github',
        },
        {
          url: 'https://www.instagram.com/ystimenkoartem',
          icon: 'instagram',
        },
      ],
    };
  },
};
</script>

<style lang="scss">
  .social {
    display: flex;
    list-style-type: none;

    &__item {
      position: relative;
      z-index: 1;
      display: inline-block;
      margin-right: 10px;

      &:last-child {
        margin-right: 0;
      }
    }
  }
</style>
