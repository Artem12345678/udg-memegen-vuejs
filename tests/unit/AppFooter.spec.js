import { shallowMount, createLocalVue } from '@vue/test-utils';
import { BootstrapVue } from 'bootstrap-vue';

import AppFooter from '@/components/AppFooter/AppFooter.vue';
import SocialList from '@/components/SocialList/SocialList.vue';

const localVue = createLocalVue();

localVue.use(BootstrapVue);

describe('AppFooter.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(AppFooter, {
      localVue,
    });
  });

  it('matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders <SocialList/> component', () => {
    expect(wrapper.find(SocialList).exists()).toBe(true);
  });
});
