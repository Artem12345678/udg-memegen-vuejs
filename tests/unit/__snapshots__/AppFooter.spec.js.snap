// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFooter.vue matches snapshot 1`] = `
<footer
  class="footer bg-secondary"
>
  <div
    class="container"
  >
    <div
      class="footer__nav"
    >
      <b-nav-stub
        tag="ul"
      >
        <b-nav-item-stub
          event="click"
          linkclasses="nav-link--footer"
          routertag="a"
          target="_self"
          to="/"
        >
          Home
        </b-nav-item-stub>
      </b-nav-stub>
       
      <div
        class="footer__social"
      >
        <sociallist-stub />
      </div>
    </div>
  </div>
</footer>
`;
